### PR TITLE
Add a method to disable all optimizations in run-time for calling thread

### PR DESCRIPTION
**Pull request by jet47**
* Add such control for tegra optimizations.
* Turn off all optimizations in CUDA tests.

### CODE REVIEW

**Comment by jet47** [comment]
@asmorkalov, @ilya-lavrenov Please check this request.

File: modules/core/include/opencv2/core/base.hpp [hunk 1]
```diff
@@ -727,6 +727,8 @@ CV_EXPORTS void setUseIPP(bool flag);
 
 } // ipp
 
+CV_EXPORTS void setUseOptimizations(bool enabled);
```
Comment (ilya-lavrenov) [inline modules/core/include/opencv2/core/base.hpp]: looks like we already have `setUseOptimized(bool)` http://docs.opencv.org/master/modules/core/doc/utility_and_system_functions_and_macros.html?highlight=setuseoptimized#cv2.setUseOptimized

Comment (jet47) [inline modules/core/include/opencv2/core/base.hpp]: Done.

**Comment by ilya-lavrenov** [comment]
:+1:

**Pull request merged**